internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -306,3 +306,37 @@ func (q amsQuotaService) DeleteQuota(subscriptionId string) *errors.ServiceError
 	}
 	return nil
 }
+
+func (q amsQuotaService) IsSubscriptionActive(kafka *dbapi.KafkaRequest) (*bool, error) {
+	billingModel, err := q.kafkaConfig.GetBillingModelByID(kafka.InstanceType, kafka.ActualKafkaBillingModel)
+	if err != nil {
+		return nil, err
+	}
+
+	orgID, err := q.amsClient.GetOrganisationIdFromExternalId(kafka.OrganisationId)
+	if err != nil {
+		return nil, err
+	}
+
+	quotas, err := q.amsClient.GetQuotaCostsForProduct(orgID, billingModel.AMSResource, billingModel.AMSProduct)
+	if err != nil {
+		return nil, err
+	}
+
+	// quota not found, subscription is not valid
+	if len(quotas) == 0 {
+		return &[]bool{false}[0], nil
+	}
+
+	// result will always return one quota for the given org, resource and product for rhosak quotas
```

Comment:
Is this correct?

If a user has consumption-based and prepaid subscriptions wouldn't we be getting more than 1 quotacost?

Reply:
Hm maybe you're right. I've only managed to test this with our organisation and I think we're only entitled prepaid. @ziccardi is there any way we can verify this?

Reply:
not really, but I believe Miguel is correct. You should check the ams_billing_model too to be sure to get one result.

Reply:
After discussions with @miguelsorianod and @ziccardi, we've reached an agreement on filtering the quota costs based on: {AMSResourceName, AMSProduct and AMSBillingModel} which should always return one quota cost. 

This works at the moment as this combination is currently unique for each quota cost but note that we may need to change this in the future depending on how quota costs are managed in AMS for RHOSAK SKUs in the future.